src/ray/util/shared_lru.h

```diff
@@ -129,7 +128,14 @@ class SharedLruCache final {
     typename std::list<Key>::iterator lru_iterator;
   };
 
-  using EntryMap = absl::flat_hash_map<Key, Entry>;
+  // TODO(hjiang): These two internal type alias has been consolidated into stable header
+  // in later versions, update after we bump up abseil.
+  using KeyHash = absl::container_internal::hash_default_hash<Key>;
+  using KeyEqual = absl::container_internal::hash_default_eq<Key>;
+
+  using KeyConstRef = std::reference_wrapper<const Key>;
```

Comment:
instead of a custom Hash and Equal functor, what about creating a MyWrapper class that contains a std::reference_wrapper<const Key> and implements Hash and Eq?

Reply:
It complicated the implementation. For example,
```
struct KeyReferenceWrapper {
  std::reference_wrapper<Key> key_ref;
  // Implement hash function: absl::container_internal::hash_default_hash<Key>
  // Implement eq function: absl::container_internal::hash_default_eq<Key>
}
```
don't see benefit.